ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/text/v1/text/clientlibs/site/js/textview.js

```diff
@@ -66,7 +66,10 @@
             updateValue(value) {
                 // html sets undefined value as undefined string in input value, hence this check is added
                 let actualValue = typeof value === "undefined" ? "" :  value;
-                const sanitizedValue = window.DOMPurify ? window.DOMPurify.sanitize(actualValue) : actualValue;
+                let sanitizedValue = window.DOMPurify ? window.DOMPurify.sanitize(actualValue, {
```

Comment:
There may be several attributes that could be filtered out by DOMPurify. Please validate all scenarios that are exposed. We are also planning to provide the complete HTML view, so DOMPurify needs to be capable of handling all these situations.

Reply:
for future reference, this should have been done before addition of DOMPurify in the code. 
Also, allowing different attributes exposes new risks for vulnerabilities, making use of DOMPurify irrelevant.
IMO, we should let customer come with a use-case and then provide a fix for it, if there is any.

Reply:
What about doing this, let's write a function speicifically for cleaning HTML in text component, since we have exposed a lot of RTE use-cases

```
// Custom configuration for DOMPurify
const cleanHTML = (dirtyHTML) => {
    // Specify the allowed tags and attributes
    const allowedTags = [
        'b', 'strong',
        'i', 'em',
        'u',
        'small',
        'blockquote',
        'code',
        'ul', 'ol', 'li',
        'a',
        'img',
        'table', 'tr', 'td', 'th',
        'h1', 'h2', 'h3', 'h4', 'h5', 'h6',
        'br', 'p'
    ];

    const allowedAttributes = {
        // Allow attributes for links
        'a': ['href', 'target', 'rel'],
        // Allow attributes for images
        'img': ['src', 'alt', 'title']
    };

    // Sanitize the HTML
    const sanitizedHTML = DOMPurify.sanitize(dirtyHTML, {
        ALLOWED_TAGS: allowedTags,
        ALLOWED_ATTR: allowedAttributes
    });

    return sanitizedHTML;
};
```